geode-core/src/main/java/org/apache/geode/cache/query/internal/DefaultQueryService.java

```diff
@@ -403,9 +403,12 @@ public Collection getIndexes(Region region) {
     if (region instanceof PartitionedRegion) {
       return ((PartitionedRegion) region).getIndexes();
     }
+
     IndexManager indexManager = IndexUtils.getIndexManager(cache, region, false);
-    if (indexManager == null)
-      return null;
+    if (indexManager == null) {
+      return Collections.emptyList();
+    }
+
```

Comment:
There are a few callers of the getIndexes(Region region) method that check for null.

That check won't be necessary anymore. You might want to remove it.

CacheXmlGenerator.generate()
DefaultQueryService.getIndexes()